### DISCUSSION TITLE
Proxy created with websocket stops working after a few minutes

### DISCUSSION THREAD

**Discussion by hayzamjs** [chimurai/http-proxy-middleware#885] [Q&A] [answered]
### Checks

- [X] I understand project setup issues should be asked on [StackOverflow](https://stackoverflow.com/questions/tagged/http-proxy-middleware) or in [GitHub Discussions](https://github.com/chimurai/http-proxy-middleware/discussions).
- [X] I updated to latest `http-proxy-middleware`.

### Describe the bug (be clear and concise)

I have an instance of ttyd that is already reverse proxied by nginx at the URL `https://192.168.1.1:4443/ttyd` and that is available and functional when I visit it. My end goal is to have this instance in an iframe in one of my web applications that uses node.js and express as the web framework.

This is how I create the proxy:

```js
const ttydProxy = createProxyMiddleware({
  target: `https://192.168.1.1:4443/ttyd`,
  changeOrigin: true,
  ws: true,
  secure: false,
  logLevel: 'silent',
  pathRewrite: {
    '^/ttyd': ''
  },
  onProxyReq: fixRequestBody
})
```

And I use it with the express app like this:

```js
app.use(['/ttyd', '/ttyd/*'], [ensureAuthenticated, ttydProxy])
```

Everything works well, atleast for a few minutes i.e before my browser starts showing this in the console with the iframe in question:

![image](https://user-images.githubusercontent.com/3922884/221379231-99a595a8-0d41-4597-bca8-b5094b7f32d9.png)

I'm not sure if it's something to do with my setup or something else entirely, would love some pointers in the right direction if it is indeed my fault

### Step-by-step reproduction instructions

Try with the config I've mentioned in the description that should be able to reproduce the issue

### Expected behavior (be clear and concise)

Websocket should be able to connect even after a long period of inactivity

### How is http-proxy-middleware used in your project?

```shell
app@0.0.1 /www/app
`-- http-proxy-middleware@2.0.6
```

### What http-proxy-middleware configuration are you using?

```typescript
const ttydProxy = createProxyMiddleware({
  target: `https://192.168.1.1:4443/ttyd`,
  changeOrigin: true,
  ws: true,
  secure: false,
  logLevel: 'silent',
  pathRewrite: {
    '^/ttyd': ''
  },
  onProxyReq: fixRequestBody
})
```

### What OS/version and node/version are you seeing the problem?

```shell
OpenWRT Linux 22.04, NodeJS v16.17.1
```

**Comment by chimurai** [comment]
If you think it's an bug; Can you provide a minimal working reproduction?

**Comment by hayzamjs** [comment]
https://codesandbox.io/s/http-proxy-middleware-forked-w6fvni?file=/src/index.js

I've made a sandbox here but I'm not sure when the actual disconnection occurs or if the sandbox will be able to catch it because it looks like the sandbox "server" reboots every now and then? 

So the issue is that when my express app starts up it is able to connect and I'm able to use ttyd well through the reverse proxy but after a while, the wss part just throws a 404 even though it was working for a while

**Comment by hayzamjs** [comment]
This is my setup now, I simplified it to just one route which is /ttyd

```js
const ttydProxy = createProxyMiddleware({
  target: `https://${getLocalIP()}:4443/ttyd`,
  changeOrigin: true,
  ws: true,
  secure: false,
  logLevel: 'debug',
  pathRewrite: {
    '^/ttyd': ''
  },
  onProxyReq: function(proxyReq, req, res) {
    //console.log('path', req.path)
    console.log('url', req.url)
  }
})
```

```js
app.use('/ttyd', [ensureAuthenticated, ttydProxy])
```

Then I initialize the express app using

```js
spdy.createServer(sslOptions, app).listen(PORT, () => {
  logger.info(`app server listening on port ${PORT}`)
});
```

I have a bunch of other stuff in `app`, I've set this up to run now, I'm gonna see if this also fails after sometime and try to get more logs out of it

EDIT: it did stop working again and I see these in my syslogs:

```
Sun Feb 26 23:05:26 2023 daemon.notice ttyd[9111]: [2023/02/26 23:05:26:2096] N: __lws_lc_untag:  -- [wsisrv|3a8|adopted] (0) 626μs
Sun Feb 26 23:05:26 2023 daemon.err nginx[12682]: 2023/02/26 23:05:26 [error] 12693#0: *31 upstream prematurely closed connection while reading response header from upstream, client: 192.168.1.1, server: _lan, request: "GET /ttyd/ttyd/ws HTTP/1.1", upstream: "http://192.168.1.1:7681//ttyd/ws", host: "192.168.1.1:4443"
Sun Feb 26 23:05:26 2023 daemon.notice ttyd[9111]: [2023/02/26 23:05:26:3549] N: __lws_lc_tag:  ++ [wsisrv|3a9|adopted] (1)
Sun Feb 26 23:05:26 2023 daemon.notice ttyd[9111]: [2023/02/26 23:05:26:3551] N: HTTP /token - 192.168.1.1
Sun Feb 26 23:05:26 2023 daemon.notice ttyd[9111]: [2023/02/26 23:05:26:3556] N: __lws_lc_untag:  -- [wsisrv|3a9|adopted] (0) 679μs
```

**Comment by hayzamjs** [comment, marked as answer]
I fixed it by removing `ws:true` from the other reverse proxies I was having, now only the ttyd proxy has `ws:true` if anyone else comes across this issue in the future, I'm not really sure why this happens, but it works for me now

**Reply by TuomasTammi** [reply]
Thank you! I had the same issue and was struggling with it for long time.
